gslib/tests/testcase/integration_testcase.py

```diff
@@ -853,35 +873,58 @@ def RunGsUtil(self, cmd, return_status=False,
       If only one return_* value was specified, that value is returned directly
       rather than being returned within a 1-tuple.
     """
-    cmd = ([gslib.GSUTIL_PATH] + ['--testexceptiontraces'] +
-           ['-o', 'GSUtil:default_project_id=' + PopulateProjectId()] +
-           cmd)
-    if IS_WINDOWS:
-      cmd = [sys.executable] + cmd
+    cmd = [
+        gslib.GSUTIL_PATH,
+        '--testexceptiontraces',
+        '-o',
+        'GSUtil:default_project_id=' + PopulateProjectId()
+    ] + cmd
+    if stdin is not None:
+      if six.PY3:
+        if isinstance(stdin, bytes):
```

Comment:
I don't think this first check is necessary -- "if it's bytes, ensure that it's really bytes" :P

Reply:
Shortened to only encode if *not* bytes.